cql3/restrictions/statement_restrictions.cc

```diff
@@ -1068,6 +1068,110 @@ std::vector<query::clustering_range> get_single_column_clustering_bounds(
     return ck_ranges;
 }
 
+// In old v1 indexes the token column was of type blob.
+// This causes problems because blobs are sorted differently than the bigint token values that they represent.
+// Tokens are encoded as 8 byte big endian two's complement signed integers,
+// which with blob sorting makes them ordered like this:
+// 0, 1, 2, 3, 4, 5, ..., bigint_max, bigint_min, ...., -5, -4, -3, -2, -1
+// Because of this clustering restrictions like token(p) > -4 and token(p) < 4 need to be translated
+// to two clustering ranges on the old index.
+std::vector<query::clustering_range> get_index_v1_token_range_clustering_bounds(
+        const query_options& options,
+        const column_definition* token_column,
+        const expression& token_restriction) {
+
+    // A workaround in order to make possible_lhs_values work properly.
+    // possible_lhs_values looks at the column type and uses this type's comparator.
+    // This is a problem because when using blob's comparator, -4 is greater than 4.
+    // This makes possible_lhs_values think that an expression like token(p) > -4 and token(p) < 4
+    // is impossible to fulfill.
+    // Create a fake token column with the type set to bigint, translate the restriction to use this column
+    // and use this restriction to calculate possible lhs values.
+    column_definition token_column_bigint = *token_column;
+    token_column_bigint.type = long_type;
+    expression new_token_restrictions = replace_column_def(token_restriction, &token_column_bigint);
```

Comment:
... it would seem that the answer is yes: every column_definition in token_restriction is token_column.

Reply:
That would require some more advanced expression manipulation. I'm not sure if it's worth it.
I am changing the calling code to remove `_uses_v1_index` so that it will include finding the token column:
```c++
    const column_definition& token_column = idx_tbl_schema.clustering_column_at(0);

    // In old indexes the token column was of type blob.
    // This causes problems with sorting and must be handled separately.
    if (token_column.type != long_type) {
        return get_index_v1_token_range_clustering_bounds(options, token_column, _idx_tbl_ck_prefix->at(0));
    }
 ```
 Might as well use it

Reply:
OK.

Reply:
It might be worth saying in the blurb something like "All binary_operators in token_restriction must have column_value{token_column} as their LHS."